File: CryptoBridge-python/src/application/services/bridge_service.py
from typing import List
from ...domain.use_cases.get_exchanges import GetExchangesUseCase
from ...domain.use_cases.get_symbols import GetSymbolsUseCase
from ...domain.use_cases.get_tickers import GetTickersUseCase
from ...domain.repositories.exchange_repository import ExchangeRepository
from ...domain.repositories.metrics_repository import MetricsRepository


class BridgeService:
    def __init__(self, exchange_repository: ExchangeRepository, metrics_repository: MetricsRepository):
        self.get_exchanges_use_case = GetExchangesUseCase(exchange_repository)
        self.get_symbols_use_case = GetSymbolsUseCase(exchange_repository, metrics_repository)
        self.get_tickers_use_case = GetTickersUseCase(exchange_repository, metrics_repository)

    async def get_exchanges(self) -> dict:
        try:
            return await self.get_exchanges_use_case.execute()
        except Exception as e:
            print(f"Erro detalhado no BridgeService.get_exchanges: {str(e)}")
            return {
                "status": 500,
                "msg": "Erro interno do servidor."
            }

    async def get_symbols(self, exchange: str) -> dict:
        try:
            return await self.get_symbols_use_case.execute(exchange)
        except Exception as e:
            print(f"Erro detalhado no BridgeService.get_symbols: {str(e)}")
            return {
                "status": 500,
                "msg": "Erro interno do servidor."
            }

    async def get_tickers(self, exchange: str, tickers: List[str]) -> dict:
        try:
            return await self.get_tickers_use_case.execute(exchange, tickers)
        except Exception as e:
            print(f"Erro detalhado no BridgeService.get_tickers: {str(e)}")
            return {
                "status": 500,
                "msg": "Erro interno do servidor."
            }

File: CryptoBridge-node/src/application/services/BridgeService.js
const GetExchangesUseCase = require('../../domain/useCases/GetExchangesUseCase');
const GetSymbolsUseCase = require('../../domain/useCases/GetSymbolsUseCase');
const GetTickersUseCase = require('../../domain/useCases/GetTickersUseCase');

class BridgeService {
    constructor(exchangeRepository, metricsRepository) {
        this.getExchangesUseCase = new GetExchangesUseCase(exchangeRepository);
        this.getSymbolsUseCase = new GetSymbolsUseCase(exchangeRepository, metricsRepository);
        this.getTickersUseCase = new GetTickersUseCase(exchangeRepository, metricsRepository);
    }

    async getExchanges() {
        try {
            return await this.getExchangesUseCase.execute();
        } catch (error) {
            console.error('Erro detalhado no BridgeService.getExchanges:', error);
            return {
                status: 500,
                msg: 'Erro interno do servidor.'
            };
        }
    }

    async getSymbols(exchange) {
        try {
            return await this.getSymbolsUseCase.execute(exchange);
        } catch (error) {
            console.error('Erro detalhado no BridgeService.getSymbols:', error);
            return {
                status: 500,
                msg: 'Erro interno do servidor.'
            };
        }
    }

    async getTickers(exchange, tickers) {
        try {
            return await this.getTickersUseCase.execute(exchange, tickers);
        } catch (error) {
            console.error('Erro detalhado no BridgeService.getTickers:', error);
            return {
                status: 500,
                msg: 'Erro interno do servidor.'
            };
        }
    }
}

module.exports = BridgeService;